src/test/e2e/psp.feature.ts

```diff
@@ -0,0 +1,353 @@
+      [threshold]
+    );
+    const beaconUpdateSubscriptionConditions = [
+      {
+        type: 'bytes32',
+        name: '_conditionFunctionId',
+        value: hre.ethers.utils.defaultAbiCoder.encode(
+          ['bytes4'],
+          [dapiServer.interface.getSighash('conditionPspBeaconUpdate')]
+        ),
+      },
+      { type: 'bytes', name: '_conditionParameters', value: beaconUpdateSubscriptionConditionParameters },
+    ];
+    const encodedBeaconUpdateSubscriptionConditions = abi.encode(beaconUpdateSubscriptionConditions);
+    await dapiServer
+      .connect(roles.randomPerson)
+      .registerBeaconUpdateSubscription(
+        airnodeWallet.address,
+        templateId,
+        encodedBeaconUpdateSubscriptionConditions,
+        airnodeWallet.address,
+        roles.sponsor.address
+      );
+
+    // Setup BTC Subscription
+    const localConfig2 = buildLocalConfig2();
+    // Templates
+    const endpointId2 = hre.ethers.utils.keccak256(
+      hre.ethers.utils.defaultAbiCoder.encode(
+        ['string', 'string'],
+        [localConfig2.endpoint.oisTitle, localConfig2.endpoint.endpointName]
+      )
+    );
+    const parameters2 = abi.encode(localConfig2.templateParameters);
+    const templateId2 = hre.ethers.utils.solidityKeccak256(['bytes32', 'bytes'], [endpointId2, parameters2]);
+
+    // Subscriptions
+    const threshold2 = (await dapiServer.HUNDRED_PERCENT()).div(localConfig2.threshold); // Update threshold %
+    const beaconUpdateSubscriptionConditionParameters2 = hre.ethers.utils.defaultAbiCoder.encode(
+      ['uint256'],
+      [threshold2]
+    );
+    const beaconUpdateSubscriptionConditions2 = [
+      {
+        type: 'bytes32',
+        name: '_conditionFunctionId',
+        value: hre.ethers.utils.defaultAbiCoder.encode(
+          ['bytes4'],
+          [dapiServer.interface.getSighash('conditionPspBeaconUpdate')]
+        ),
+      },
+      { type: 'bytes', name: '_conditionParameters', value: beaconUpdateSubscriptionConditionParameters2 },
+    ];
+    const encodedBeaconUpdateSubscriptionConditions2 = abi.encode(beaconUpdateSubscriptionConditions2);
+    await dapiServer
+      .connect(roles.randomPerson)
+      .registerBeaconUpdateSubscription(
+        airnodeWallet.address,
+        templateId2,
+        encodedBeaconUpdateSubscriptionConditions2,
+        airnodeWallet.address,
+        roles.sponsor.address
+      );
+  });
+
+  it('updates the beacon successfully', async () => {
+    jest
+      .spyOn(config, 'loadAirnodeConfig')
+      .mockImplementationOnce(() => airnodeConfig as any)
+      .mockImplementationOnce(() => airnodeConfig as any);
+    jest.spyOn(config, 'loadAirkeeperConfig').mockImplementationOnce(() => airkeeperConfig as any);
+    const res = await psp.handler();
+
+    const voidSigner = new hre.ethers.VoidSigner(hre.ethers.constants.AddressZero, provider);
+
+    // ETH price beacon
+    const beaconIdETH = await dapiServer.subscriptionIdToBeaconId(subscriptionIdETH);
```

Comment:
Maybe create a helper function for reading a beacon value? That is repeated multiple times in the tests.